src/filters/time/time.ts

```diff
@@ -16,9 +16,54 @@ export default class Time {
         });
     }
 
+    static convertVerbalExpressionToObj(matchingText: string): TimeObjectSchema {
```

Comment:
I might break this logic out / up.

Reply:
Can't do any more work on this today, so leaving it out there for an opinion.

Reply:
Yeah simplify the logic here, break it down. I also think we should move this to utils + test it

Reply:
Well, the logic is tested already, I guess we can break it off and unit test it specifically. At the moment it is only used in one place though, so I am not too sure if there is a lot of value of moving this to utils, since it is getting a little bit bloated, don't you think?

Reply:
I'd say a function with that much logic should be unit tested on its own. I know there are high level tests, but those aren't unit tests for this function even if it does touch this. Tbh we should be mocking more functions. We are kind of doing integration testing in our unit tests at the moment.